### ISSUE TITLE
Rule proposal: JSDoc relaxation

### ISSUE THREAD

**Issue by mathieumg** [yannickcr/eslint-plugin-react#76]
ESLint has the [valid-jsdoc](https://github.com/eslint/eslint/blob/master/docs/rules/valid-jsdoc.md) rule which allows us to validate that certain JSDoc elements are present, when relevant. However, it complains when the React `render` method does not provide a return value in the JSDoc. i.e.

```javascript
/**
 * React render
 */
render: function() {
  //...
}
```

This is just a nit and I have no idea how much power, if at all, eslint-plugin-react has towards the application of this rule by eslint, but is there any conceivable way in which it would prevent valid-jsdoc from emitting that error, there, for all React components? Sort of like, toggling the `requireReturn` to false if the method is called `render`.

Thank you!

**Comment by yannickcr** [comment]
Sadly, as far as I know, a plugin don't have the power to silent an error reported by an ESLint rule, the only way we have to do something approaching is to replicate the ESLint rule, "patch" it to fit our needs and use it instead of the ESLint one. It is not very clean and it can be time consuming to maintain the rule in sync with the ESLint one.

Also, about your case, I do not understand why you do not want to document the return of render. In my opinion it should be documented like any other function.

**Comment by AlexKVal** [comment]
> it can be time consuming to maintain the rule in sync with the ESLint one.

Agree.

It seems that adding simple `@returns smth` line to `render()` is much simpler solution.

**Comment by mathieumg** [comment]
> Also, about your case, I do not understand why you do not want to document the return of render. In my opinion it should be documented like any other function.

You are right, but I find it would be cumbersome and repetitive to always document React's `render` as it's always gonna be the same. That being said, I think the fault is mine, I shouldn't have put that doc-block-looking comment immediately before the render method, I'll add an empty space in-between like I have done for other sections of my components. ("React properties", "React lifecycle", "Own methods", etc.)

Finally, thanks for confirming my doubts regarding the limitations of the plugin system!
